x-pack/plugins/enterprise_search/common/ml_inference_pipeline/index.ts

```diff
@@ -11,6 +11,7 @@ import {
   MlTrainedModelConfig,
   MlTrainedModelStats,
 } from '@elastic/elasticsearch/lib/api/types';
+import { ElasticModels } from '@kbn/ml-plugin/public/application/services/elastic_models_service';
```

Comment:
you should use
```suggestion
import { type MlPluginStart } from '@kbn/ml-plugin/public';
```
Also `common` isn't a suitable location for that. You'll need to place it somewhere under the `public` dir of your plugin.

Reply:
I moved this code in the TextExpansionCalloutLogic which is part of public (plus it is a better choice).